server/src/main/java/org/elasticsearch/transport/OutboundMessage.java

```diff
@@ -41,20 +42,39 @@
     BytesReference serialize(BytesStreamOutput bytesStream) throws IOException {
         storedContext.restore();
         bytesStream.setVersion(version);
-        bytesStream.skip(TcpHeader.HEADER_SIZE);
+        bytesStream.skip(TcpHeader.headerSize(version));
 
         // The compressible bytes stream will not close the underlying bytes stream
         BytesReference reference;
+        int variableHeaderLength = -1;
+        final long preHeaderPosition = bytesStream.position();
+
+        boolean needToWriteHeader = true;
```

Comment:
should we just drop this boolean and instead check `if (variableHeaderLength != -1)`?